climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+        temp = pd.DataFrame()
+        if 'instrument' in firms.columns:
+            if firms.instrument.any() == 'MODIS' or firms.instrument.any() == 'VIIRS':
+                firms_modis = firms.drop(firms[firms.instrument == 'VIIRS'].index)
+                firms_modis.confidence = np.array(
+                    list(map(int, firms_modis.confidence.values.tolist())))
+                firms_modis = firms_modis.drop(firms_modis[ \
+                    firms_modis.confidence < CLEAN_THRESH].index)
+                temp = firms_modis
+                firms_viirs = firms.drop(firms[firms.instrument == 'MODIS'].index)
+                if firms_viirs.size:
+                    firms_viirs = firms_viirs.drop(firms_viirs[firms_viirs.confidence == 'l'].index)
+                    firms_viirs = firms_viirs.rename(columns={'bright_ti4':'brightness'})
+                    temp = temp.append(firms_viirs, sort=True)
+                    temp = temp.drop(columns=['bright_ti4'])
+
+            firms = temp
+            firms = firms.reset_index()
+            firms = firms.drop(columns=['index'])
+
+        firms['iter_ev'] = np.ones(len(firms), bool)
+        firms['cons_id'] = np.zeros(len(firms), int) - 1
+        firms['event_id'] = np.zeros(len(firms), int)
+        firms['clus_id'] = np.zeros(len(firms), int) - 1
+        firms['datenum'] = np.array(u_dt.str_to_date(firms['acq_date'].values))
+        return firms
+
+    @staticmethod
+    def _firms_resolution(firms):
+        """ Returns resolution of satellite used in FIRMS in degrees
+
+        Parameters
+        ----------
+            csv_firms : pd.DataFrame or str
+                 path to csv file of FIRMS data or FIRMS data as pd.Dataframe
+
+        Returns
+        -------
+            res_data/ONE_LAT_KM : float
+                resolution in degrees
+
+        """
+        # Resolution in km of the centroids depends on the data origin.
+        if 'instrument' in firms.columns:
+            if firms['instrument'].any() == 'MODIS':
+                res_data = 1.0
+            else:
+                res_data = 0.375 # For VIIRS data
+        else:
+            res_data = RES_DATA # For undefined data origin, defined by user
+        return res_data/ONE_LAT_KM
+
+    @staticmethod
+    def _firms_centroids_creation(firms, res_data, centr_res_factor):
+        """ Get centroids from the firms dataset and refactor them.
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+            res_data : float
+                FIRMS instrument resolution in degrees
+            centr_res_factor : float
+                the factor applied to voluntarly decrease/increase the
+                centroids resolution
+
+        Returns
+        -------
+            centroids : Centroids
+
+        """
+        centroids = Centroids()
+        centroids.set_raster_from_pnt_bounds((firms['longitude'].min(), \
```

Comment:
This method will not be able to deal correctly with data crossing the antimeridian but not containing the whole world. This is particularly annoying when for instance considering the USA.

Reply:
agreed. However, for countries that have overseas territories it doesn't make sense to calculate everything in one grid, as this would blow up the raster massively without any use. As one single fire typically doesn't cross the anti-meridian (in contrast to TCs) I don't consider this a big problem here.